doc/timing.adoc

```diff
@@ -0,0 +1,259 @@
+==== Usage
+[source,lua]
+----
+minetest.register_globalstep(function(dtime)
+	-- do something, probably involving dtime
+end)
+----
+
+==== Params
+[%autowidth, frame=none]
+|===
+| `dtime` | number | Delta (elapsed) time since last step in seconds
+|===
+
+TIP: Use globalsteps to poll for an event for which there are no callbacks, such as player controls (`player:get_player_control()`).
+
+CAUTION: As globalsteps run every server step, they are highly performance-critical and must be well optimized. If globalsteps take longer than the server step to complete, the server thread is blocked and the server becomes "laggy".
+
+==== Examples
+
+Globalsteps are ideal to run something periodically, which is often used to call expensive operations less frequently in order to keep server thread blockage - and thus lag - to a minimum.
+
+[source,lua]
+----
+-- period is in seconds
+local function run_periodically(period, func)
+	local timer = 0
+	minetest.register_globalstep(function(dtime)
+		timer = timer + dtime
+		if timer > period then
+			func()
+			timer = 0
+		end
+	end)
+end
+run_periodically(5, function()
+	minetest.chat_send_all("5 seconds have passed!")
+end)
+----
+
+This will call `func` after at least `timer` seconds have passed. Note that there is no "catchup": The timer is always reset to zero, no matter how "late" the call to func is - how large `timer - period` is. For catchup, you can simply set the timer to the leftover time instead: `timer = timer - period`. This will trigger calls in subsequent steps until the missed time is "catched up".
+
+=== `minetest.after(time, func, ...)`
+
+Will call `func(...)` after at least `time` seconds have passed.
+
+TIP: Use `minetest.after(0, func)` to immediately do load-time stuff that is only possible at run-time, or to schedule something for the next server step.
+
+[NOTE]
+.Order of execution
+====
+Scheduled calls happen in reverse insertion order. This "defers" calls to `minetest.after` from within a callback until the next server step. Examples:
+
+[source,lua]
+----
+minetest.after(0, error, "first")
+minetest.after(0, error, "second")
+----
+This script will error with `second` after the first server step,
+as the callbacks that are scheduled to run in the same server step are processed in reverse order.
+
+The following snippet emulates a globalstep, sending `Hello World!` to chat every server step:
+[source,lua]
+----
+local function after_step()
+	minetest.chat_send_all("Hello World!")
+	minetest.after(0, after_step)
+end
+minetest.after(0, after_step)
+----
+====
+
+==== Usage
+[source,lua]
+----
+job = minetest.after(time, func, ...)
+if ... then
+	job:cancel()
+end
+----
+
+==== Params
+[%autowidth, frame=none]
+|===
+| `time` | number | Time in seconds that must have passed for the callback to be executed.
+| `func` | function | Function to be called. Objects with the `__call` metamethod are supported as well.
+| `...` | vararg | Arguments to be passed to `func` when it is called.
```

Comment:
We should add a `{type-vararg}` variable/macro/substitution to types.adoc

Reply:
This can be done later, as it's not directly related to the actual meat of this PR.